embedded-tests/src/test/java/org/apache/druid/testing/embedded/query/JdbcQueryTest.java

```diff
@@ -145,84 +124,91 @@ public void testJdbcMetadata()
   {
+    String query = StringUtils.format(QUERY_TEMPLATE, tableName, "1000");
     for (String url : connections) {
       try (Connection connection = DriverManager.getConnection(url, connectionProperties)) {
         try (Statement statement = connection.createStatement()) {
-          final ResultSet resultSet = statement.executeQuery(QUERY);
+          final ResultSet resultSet = statement.executeQuery(query);
           int resultRowCount = 0;
           while (resultSet.next()) {
             resultRowCount++;
             LOG.info("%s,%s,%s", resultSet.getString(1), resultSet.getLong(2), resultSet.getLong(3));
           }
-          Assert.assertEquals(resultRowCount, 10);
+          Assertions.assertEquals(7, resultRowCount);
           resultSet.close();
         }
       }
       catch (SQLException throwables) {
-        Assert.fail(throwables.getMessage());
+        Assertions.fail(throwables.getMessage());
       }
     }
   }
 
   @Test
   public void testJdbcPrepareStatementQuery()
   {
+    String query = StringUtils.format(QUERY_TEMPLATE, tableName, "?");
     for (String url : connections) {
       try (Connection connection = DriverManager.getConnection(url, connectionProperties)) {
-        try (PreparedStatement statement = connection.prepareStatement(QUERY_PARAMETERIZED)) {
-          statement.setString(1, "en");
+        try (PreparedStatement statement = connection.prepareStatement(query)) {
+          statement.setLong(1, 1000);
           final ResultSet resultSet = statement.executeQuery();
           int resultRowCount = 0;
           while (resultSet.next()) {
             resultRowCount++;
             LOG.info("%s,%s,%s", resultSet.getString(1), resultSet.getLong(2), resultSet.getLong(3));
           }
-          Assert.assertEquals(resultRowCount, 10);
+          Assertions.assertEquals(7, resultRowCount);
           resultSet.close();
         }
       }
       catch (SQLException throwables) {
-        Assert.fail(throwables.getMessage());
+        Assertions.fail(throwables.getMessage());
       }
     }
   }
 
-  @Test(expectedExceptions = AvaticaSqlException.class, expectedExceptionsMessageRegExp = ".* No value bound for parameter \\(position \\[1]\\)")
-  public void testJdbcPrepareStatementQueryMissingParameters() throws SQLException
+  @Test
+  public void testJdbcPrepareStatementQueryMissingParameters()
   {
+    String query = StringUtils.format(QUERY_TEMPLATE, tableName, "?");
     for (String url : connections) {
       try (Connection connection = DriverManager.getConnection(url, connectionProperties);
-           PreparedStatement statement = connection.prepareStatement(QUERY_PARAMETERIZED);
+           PreparedStatement statement = connection.prepareStatement(query);
            ResultSet resultSet = statement.executeQuery()) {
         // This won't actually run as we expect the exception to be thrown before it gets here
         throw new IllegalStateException(resultSet.toString());
```

Comment:
Should we use something like `Assertions.fail()` instead?

Reply:
https://github.com/apache/druid/pull/18805/commits/0a64c9bc05d01424463e0897a8f069027e12c7a2